bank/src/components/Breakdown.js

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from "react"
+import Category from "./Category"
+import Balance from "./Balance"
+import "../styles/breakdown.css"
+
+class Breakdown extends Component {
+  setAsActiveTab = () => {
+    this.props.setActiveTab("breakdown")
+  }
+  render() {
+    if (!this.props.isActiveTab) {
+      this.setAsActiveTab()
+    }
+    const categories = []
+    this.props.transactions.forEach((transaction) => {
+      if (!categories.includes(transaction.category.toLowerCase())) {
+        categories.push(transaction.category.toLowerCase())
+      }
+    })
+    return (
+      <div id="breakdown-page">
+        <Balance balance={this.props.balance} />
+        <h2>Transactions Breakdown</h2>
+        {categories.length ? (
+          <div id="categories">
+            <table>
+              <tbody>
+                <tr>
+                  <th className="category-title">Category</th>
+                  <th className="total">Total</th>
+                  <th>Report</th>
+                </tr>
+                {categories.map((category, index) => {
+                  let total = 0,
+                    transactions = []
+                  this.props.transactions.forEach((t) => {
+                    if (t.category.toLowerCase() === category) {
+                      transactions.push(t)
+                      total += t.amount
+                    }
```

Comment:
a really good practice is to move this to an outside function that would calculate the total, and not do it here in the render